jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -673,11 +684,109 @@ public void write(int b) throws IOException
 
     @Override
     public void print(String s) throws IOException
+    {
+        print(s,false);
+    }
+
+    @Override
+    public void println(String s) throws IOException
+    {
+        print(s,true);
+    }
+
+    private void print(String s, boolean eoln) throws IOException
     {
         if (isClosed())
             throw new IOException("Closed");
 
-        write(s.getBytes(_channel.getResponse().getCharacterEncoding()));
+        // TODO would a Threadlocal pool be better?
+        String charset = _channel.getResponse().getCharacterEncoding();
+        if (_encoder==null || !_encoder.charset().name().equals(charset))
+            _encoder = Charset.forName(charset).newEncoder();
+        else
+            _encoder.reset();
```

Comment:
It is not thread safe to store `_encoder` as a field.
We know that application can write concurrently, and we know that the encoder is stateful, so I'm for creating the encoder on the fly and GC it at end of method.

Reply:
It caches them in String.getBytes implementation, which is not the path we use.  So this code does not use their ThreadLocalEncoderCache

Reply:
I've gone with a simple ThreadLocal cache of a single encoder.  Not as good as the JVMs MRU array, but simpler and better than no cache in many circumstances